### PR TITLE
Documentation: clean up and added examples

### PR DESCRIPTION
**Pull request by gonzalocasas**
image

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [x] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_eve/core.py [hunk 1]
```diff
@@ -8,10 +8,25 @@
 
 
 def get_default_transport():
+    """Retrieve the default, system-wide transport.
+
+    Returns
+    -------
+    :class:`~compas_eve.Transport`
+        _description_
```
Comment (chenkasirer) [inline src/compas_eve/core.py:16]: what does this do?

File: src/compas_eve/core.py [hunk 2]
```diff
@@ -64,7 +79,21 @@ def parse(cls, value):
...
+    set of options.
+
+    Attributes
+    ----------
+    name : str
+        Name of the topic.
+    message_type : type
+        Class defining the message structure. Use :class:`Message` for
+        a generic, non-typed checked message implementation.
+    options : dict
```
Comment (chenkasirer) [inline src/compas_eve/core.py:94]: maybe an example of some options in `examples`?

File: src/compas_eve/mqtt/mqtt_paho.py [hunk 3]
```diff
@@ -68,14 +114,21 @@ def _on_message(self, client, userdata, msg):
         self.emit(event_key, msg)
 
     def advertise(self, topic):
+        """Announce this code will publish messages to the specified topic.
```
Comment (chenkasirer) [inline src/compas_eve/mqtt/mqtt_paho.py:117]: missing `Parameters`?

**Review by chenkasirer — APPROVED** [review]
LGTM

File: src/compas_eve/core.py (hunk 1 above)
Comment (gonzalocasas) [reply src/compas_eve/core.py]: fixed :)

File: src/compas_eve/core.py (hunk 2 above)
Comment (gonzalocasas) [reply src/compas_eve/core.py:95]: right now, there are no options implemented, so, I just added a `TODO` note https://github.com/compas-dev/compas_eve/pull/4/commits/42a33e53d36a57ed517feb640d4f6f15dfa25830

File: src/compas_eve/mqtt/mqtt_paho.py (hunk 3 above)
Comment (gonzalocasas) [reply src/compas_eve/mqtt/mqtt_paho.py:117]: done

**Pull request merged**